lib/helpers.ts

```diff
@@ -0,0 +1,348 @@
+import Exception from './exception.js';
+import type {
+  CloseBlock,
+  InverseChain,
+  LocInfo,
+  OpenBlock,
+  OpenPartialBlock,
+  OpenRawBlock,
+  Part,
+  SourcePosition,
+} from './types/types.js';
+import type * as ast from './types/ast.js';
+import { assert } from './utils.js';
+import type { ParseOptions, SyntaxOptions } from './parse.js';
+
+export class ParserHelpers {
+  #options: ParseOptions;
+  readonly syntax: SyntaxOptions;
+
+  constructor(options: ParseOptions) {
+    this.#options = options;
+
+    let squareSyntax: SyntaxOptions['square'];
+
+    if (typeof options?.syntax?.square === 'function') {
```

Comment:
```suggestion
    if (typeof options.syntax?.square === 'function') {
```
options themselves aren't optional here.

Otherwise you'd want `options?: ParseOptions` on the constructor

Reply:
Fixed, but I feel like this should be a lint error. Probably an indication that we should beef up the linting setup now that the `lib` directory is all TypeScript.